create_session_parquet_files.py

```diff
@@ -0,0 +1,61 @@
+import os
+import argparse
+import re
+
+from astropy.table import Table, vstack
+import pandas as pd
+import numpy as np
+
+# regular expression to match the session name given a path to antenna FITS file
+SESSION_REGEX = re.compile(r"/[^/]*/Antenna")
+
+def group_files_df(manifest_path: str):
+    """Returns a pd dataframe with one column containing all the paths listed in the manifest file, and
+    another column containing each file's associated session"""
+    lines = open(manifest_path, "r").readlines()
+    df = pd.DataFrame(lines, columns=["path"])
+    df["path"] = df["path"].map(lambda x: x.replace("\n", ""))
+    # get session name
+    df["session"] = df["path"].map(lambda x: re.search(SESSION_REGEX, x).group(0).split("/")[1])
+    return df
```

Comment:
This function iterates over every row in the file at least 3 times. It would be better to do everything in a single pass, e.g.

```suggest
def group_files_df(manifest_path: Path | str):
    """Create DataFrame containing path and session for each file referenced in given manifest"""
    with open(manifest_path, "r") as file:
        rows = ((path.rstrip("\n"), Path(path).parent.parent.name) for path in file)
        return pd.DataFrame(rows, columns=["path", "session"])
```